internal/clients/http/sources/sources_client.go

```diff
@@ -90,7 +92,16 @@ func (c *sourcesClient) ListProvisioningSourcesByProvider(ctx context.Context, p
 		return nil, fmt.Errorf("failed to get provisioning app type: %w", err)
 	}
 
-	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddSourcesIdentityHeader, headers.AddEdgeRequestIdHeader)
+	sourcesProviderName, err := c.getProviderNameBySources(ctx, provider)
+	if err != nil {
+		logger.Warn().Err(err).Msg("Failed to get provider name according to sources service")
+		return nil, fmt.Errorf("failed to get provider name according to sources service: %w", err)
+	}
+
+	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddSourcesIdentityHeader, headers.AddEdgeRequestIdHeader, func(ctx context.Context, req *stdhttp.Request) error {
+		req.URL.RawQuery = url.QueryEscape("filter[source_type][name]") + fmt.Sprintf("=%s", sourcesProviderName)
+		return nil
+	})
```

Comment:
Can you extract this inline function into named function that will return `func(ctx context.Context, req *stdhttp.Request)` with variadic string arguments, so any number of query params would work? I think this will be limitation across all endpoints so we might want to use it. Also pagination will be added soon.

This should look like something:

```
c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddSourcesIdentityHeader, headers.AddEdgeRequestIdHeader, xxx.BuildQuery("filter[source_type][name]", sourcesProviderName))
```

In the future this could be `xxx.BuildQuery("filter[source_type][name]", sourcesProviderName, "limit", limit, "offset", offset)`